src/gui/editors/PianoRoll.cpp

```diff
@@ -4357,15 +4357,17 @@ void PianoRollWindow::stop()
 
 void PianoRollWindow::record()
 {
-	m_editor->record();
+	Engine::getSong()->startRecordCountDown();
+	QTimer::singleShot( Engine::getSong()->getRecordDelay() * 1000 , m_editor, SLOT( record() ) );
```

Comment:
Since the countdown will be based on bars and the metronome is going to play, you should rely on the rendering process and tick count instead of timers.